Sources/StreamChat/ChatClient.swift

```diff
@@ -378,6 +378,21 @@ public class ChatClient {
         }
     }
 
+    /// Get the link attachment preview data from the provided url.
+    ///
+    /// This will return the data present in the OG Metadata.
+    public func enrichUrl(_ url: URL, completion: @escaping (Result<LinkAttachmentPayload, Error>) -> Void) {
```

Comment:
I don't think this should be in the `ChatClient` - we mostly have `connect` / `disconnect` methods here. Maybe in the channel controller?

Reply:
This is to be aligned with the other SDKs, they also have `enrichUrl` in the `ChatClient`. That is why I opted for putting it in the `ChatClient`. I guess we can also move it to the `ChatChannelController`, but enriching a URL is not dependent on a Channel, so if we move the `enrichUrl` to the `CHatChannelController` it means the customer needs to create a controller or load a channel just to enrich a url

Reply:
hmm, do all SDKs have it in their client or some specific one? In general, the preview is a composer feature, so enriching URLs by only being part of a channel is not an issue. However, I think the decision tree should be as follows - if more than 1 SDK has it in the client, then leave it as it is for consistency, otherwise move it to the channel controller. What do you think?

Reply:
I can see that `uploadAttachment` is also part of `ChatClient` in other SDKs. All SDKs have this. But we have `uploadAttachment` in `ChannelController` as well. So maybe moving it to `ChannelController` makes it more consistent on our SDK, it is also just more straightforward to use I guess. But this will require some changes in tests etc...

Reply:
@martinmitrevski Moved to the ChannelController 👍